NeuroID/src/test/java/com/neuroid/tracker/NeuroIDClassUnitTests.kt

```diff
@@ -1222,4 +1224,90 @@ open class NeuroIDClassUnitTests {
 
         assertWarningCount(1)
     }
+
+    @Test
+    fun testStartAppFlow_no_userID_session_started() = runTest {
+        mockkStatic(Calendar::class)
+        every{Calendar.getInstance().timeInMillis} returns 0
+        setMockedNIDJobServiceManager(false)
+        setMockedDataStore()
+        NeuroID.getInternalInstance()?.let {
+            it.clientKey = "dummyKey"
+            val (started, id) = it.startSession("test12434")
+            val flowResult = it.startAppFlow("form_zzzzz123")
+            assertEquals(true, started)
+            assertEquals("test12434", id)
+            assertEquals(true, flowResult.started)
+            assertEquals("", flowResult.sessionID)
+            verify {NeuroID.getInternalInstance()?.nidJobServiceManager?.sendEvents(
+                true)}
+            verify {NeuroID.getInternalInstance()?.dataStore?.saveEvent(NIDEventModel(
+                type= SET_LINKED_SITE, v="form_zzzzz123"))}
+            assertEquals("form_zzzzz123", NeuroID.linkedSiteID)
+        }
+        unmockkStatic(Calendar::class)
+    }
+
+    @Test
+    fun testStartAppFlow_userID_session_not_started() = runTest {
+        mockkStatic(Calendar::class)
+        every{Calendar.getInstance().timeInMillis} returns 0
+        setMockedNIDJobServiceManager(false)
+        setMockedDataStore()
+        NeuroID.getInternalInstance()?.let {
+            it.clientKey = "dummyKey"
+            val flowResult = it.startAppFlow("form_zzzzz123", "test_1234")
+            assertEquals(true, flowResult.started)
+            assertEquals("test_1234", flowResult.sessionID)
+            assertEquals(false, NeuroID.getInstance()?.isStopped())
+            verify {NeuroID.getInternalInstance()?.nidJobServiceManager?.sendEvents(
+                true)}
+            verify {NeuroID.getInternalInstance()?.dataStore?.saveEvent(NIDEventModel(
+                type= SET_LINKED_SITE, v="form_zzzzz123"))}
+            assertEquals("form_zzzzz123", NeuroID.linkedSiteID)
+        }
+        unmockkStatic(Calendar::class)
+    }
```

Comment:
this test might be named wrong. it says session not started but we assert that it is? I don't see us setting isSDKStarted to true prior to the test so presumably startSession is happening?